toolkit/tools/pkg/imagecustomizerlib/shrinkfilesystems.go

```diff
@@ -129,24 +136,40 @@ func getFilesystemSizeInSectors(resize2fsOutput string, imageLoopDevice string)
 	}
 	sectorSizeInBytes := int(logicalSectorSize) // cast from uint64 to int
 
-	return int(math.Ceil(float64(filesystemSizeInBytes) / float64(sectorSizeInBytes))), nil
+	filesystemSizeInSectors = filesystemSizeInBytes / sectorSizeInBytes
+	if filesystemSizeInBytes%sectorSizeInBytes != 0 {
+		filesystemSizeInSectors++
+	}
+
+	return filesystemSizeInSectors, nil
 }
 
 // Get the new partition end in sectors.
 func getNewPartitionEndInSectors(resize2fsOutput string, startSector string, imageLoopDevice string) (endInSectors string, err error) {
 	filesystemSizeInSectors, err := getFilesystemSizeInSectors(resize2fsOutput, imageLoopDevice)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get filesystem size: \n%w", err)
 	}
 
 	// Convert start sector string to int
 	start, err := strconv.Atoi(startSector)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to convert start sector to int: \n%w", err)
 	}
 	// Calculate the new end
 	end := start + filesystemSizeInSectors
 	// Convert to a string with sectors unit appended
 	endInSectors = strconv.Itoa(end) + "s"
 	return endInSectors, nil
 }
+
+// Checks if the provided fstype is supported.
+func supportedFsType(fstype string) (isSupported bool) {
```

Comment:
Functions are private to the module, not the file. So, this name is a little too generic.

Reply:
good point, updated